File: tests/test_insertion_sort.py
from random import randint

from sys import path
path.append('..')

from insertion_sort import insertion_sort

def test_insertion_sort():
    test_list = [randint(1, 50) for i in range(15)]
    test_sorted = insertion_sort(test_list[:])
    if test_sorted == sorted(test_list):
        print("test passes")
        print(test_list)
        print(test_sorted)
    else:
        print("test fails")
        print(test_list)
        print(test_sorted)

if __name__ == "__main__":
    test_insertion_sort()

File: tests/TestInsertionSort.java
package Standard_Algorithms.tests;

import Standard_Algorithms.InsertionSort;
import java.util.Arrays;

public class TestInsertionSort
{
    public static void main(String[] args)
    {
        int[] testArray = new int[12];
        for (int i=0; i<12; i++)
            {
                testArray[i] = (int)(50*Math.random());
            }

        int[] testSorted = Arrays.copyOf(testArray, 12);
        InsertionSort.insertionSort(testSorted);

        int[] testLibrarySorted = Arrays.copyOf(testArray, 12);
        Arrays.sort(testLibrarySorted);

        if (!Arrays.equals(testSorted, testLibrarySorted))
            System.out.println("TEST FAILS");
        else
            System.out.println("test passes");
    }

}